plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -826,13 +843,47 @@ public List<SchemaTableName> listTables(ConnectorSession session, Optional<Strin
     @Override
     public Map<SchemaTableName, RelationType> getRelationTypes(ConnectorSession session, Optional<String> optionalSchemaName)
     {
-        ImmutableMap.Builder<SchemaTableName, RelationType> result = ImmutableMap.builder();
-        for (String schemaName : listSchemas(session, optionalSchemaName)) {
-            for (TableInfo tableInfo : metastore.getTables(schemaName)) {
-                result.put(tableInfo.tableName(), tableInfo.extendedRelationType().toRelationType());
+        List<Callable<List<TableInfo>>> tasks = listSchemas(session, optionalSchemaName).stream()
+                .map(schemaName -> (Callable<List<TableInfo>>) () -> metastore.getTables(schemaName))
+                .collect(toImmutableList());
+        return runInParallel(tasks).stream()
+                .collect(toImmutableMap(TableInfo::tableName, tableInfo -> tableInfo.extendedRelationType().toRelationType(), (ignore, second) -> second));
+    }
+
+    private List<TableInfo> runInParallel(List<Callable<List<TableInfo>>> tasks)
```

Comment:
Can we move `io.trino.plugin.hive.metastore.glue.ExecutorUtil#processWithAdditionalThreads` to plugin-toolkit and reuse ?